nnpdf_data/nnpdf_data/new_commondata/STAR_2015_1JET_200GEV/filter.py

```diff
@@ -0,0 +1,154 @@
+
+            df3["eta_min"] = -0.5
+            df3["eta_max"] = 0.5
+            df3["eta"] = 0.0
+            df3["sqrts"] = 200
+
+        elif "4" in fname:
+            dfi = pd.read_csv(fname)
+            if "pT" in fname:
+                correlation = False
+                df4["pT"] = dfi["Parton Jet $p_T$ [GeV/c]"]
+
+            if "ALL" in fname:
+                df4["ALL"] = dfi["$A_{LL}$"]
+                df4["stat_min"] = dfi["stat +"]
+                df4["stat_max"] = dfi["stat -"]
+                df4["sys_min"] = dfi["sys +"]
+                df4["sys_max"] = dfi["sys -"]
+
+            df4["eta_min"] = -1.0
+            df4["eta_max"] = 1.0
+            df4["eta"] = 0.0
+            df4["sqrts"] = 200
+
+        elif "5" in fname:
+            correlation = True
+            dfc_col = pd.read_csv(fname)
+
+            dfc = pd.DataFrame()
+            biny = 1
+            for i in range(len(dfc_col)):
+                if dfc_col.loc[i, "binx"] == "-":
+                    biny = float(dfc_col.loc[i, "val"])
+                else:
+                    binx = float(dfc_col.loc[i, "binx"])
+                    dfc.loc[binx, biny] = dfc_col.loc[i, "val"]
+
+            dfc = dfc.astype("float")
+
+        else:
+            print("ERROR: Unknown table number detected! Check input files.")
+
+    df = pd.concat([df3, df4], ignore_index=True)
+
+    for i in range(len(df)):
+        shift, unc = symmetrize_errors(df.loc[i, "stat_max"], df.loc[i, "stat_min"])
+        df.loc[i, "stat"] = unc
+        df.loc[i, "ALL"] += shift
+
+        shift, unc = symmetrize_errors(df.loc[i, "sys_max"], df.loc[i, "sys_min"])
+        df.loc[i, "sys"] = unc
+        df.loc[i, "ALL"] += shift
+
+    return df, dfc
+
+
+def write_data(df, dfc):
+    data_central = []
+    for i in range(len(df["ALL"])):
+        data_central.append(float(df.loc[i, "ALL"]))
+
+    data_central_yaml = {"data_central": data_central}
+    with open("data.yaml", "w") as file:
+        yaml.dump(data_central_yaml, file, sort_keys=False)
+
+    # Compute the covariance matrix
+    corrmat = dfc.values.reshape((len(df), len(df)))
+    art_sys = compute_covmat(corrmat, df["stat"], len(df))
+
+    # Write kin file
+    kin = []
+    for i in range(len(df["ALL"])):
+        kin_value = {
+            "pT": {"min": None, "mid": float(df.loc[i, "pT"]), "max": None},
+            "sqrts": {"min": None, "mid": float(df.loc[i, "sqrts"]), "max": None},
+            "eta": {
+                "min": float(df.loc[i, "eta_min"]),
+                "mid": float(df.loc[i, "eta"]),
+                "max": float(df.loc[i, "eta_max"]),
+            },
+        }
+        kin.append(kin_value)
+
+    kinematics_yaml = {"bins": kin}
+
+    with open("kinematics.yaml", "w") as file:
+        yaml.dump(kinematics_yaml, file, sort_keys=False)
+
+    # Write unc file
+    error = []
+    for i in range(len(df)):
+        e = {"stat": float(df.loc[i, "stat"]), "sys": float(df.loc[i, "sys"])}
```

Comment:
Here the situation is a bit more subtle.
My understanding is that the correlated matrix is for syst and stat all together, so you need to do the decomposition and replace them by the artificial syst (see the hermes data).
But the hepdata table also says that we need to add: 

 ```
 In addition to the point-to-point statistical and systematic uncertainties, there are two systematic uncertainties that are common to all the measured A_LL values (and not included in the systematic uncertainty numbers quoted above): (a)  +/-0.0005 vertical shift uncertainty from relative luminosity and (b)  +/-6.5% vertical scale uncertainty from polarization.
 ```
 
 for all the datapoints.

Reply:
so in what way should these two extra uncertainties be added? For one: what should they be called? @giacomomagni

Reply:
* Luminosity --> `lumi` additive uncorrelated
* polarization --> `pol` multiplicative correlated